contract/src/lib.rs

```diff
@@ -1,162 +1,113 @@
+use near_sdk::{near_bindgen, setup_alloc};
 
 setup_alloc!();
 
 #[derive(Serialize, Deserialize, BorshDeserialize, BorshSerialize, Debug, Clone)]
 #[serde(crate = "near_sdk::serde")]
-pub struct FN{
-    pub id:u32,
+pub struct News
+{
+    pub id: u32,
     pub hash_head: String,
     pub hash_body: String,
     pub uri: String,
     pub like: u64,
     pub dislike: u64,
 }
 
-
-
-
 #[near_bindgen]
-#[derive(BorshDeserialize, BorshSerialize, Debug, Clone)]
-pub struct News{
-    news: Vec<FN>,
+#[derive(Default, BorshDeserialize, BorshSerialize, Debug, Clone)]
+pub struct NewsStorage
+{
+    news: Vec<News>,
 }
 
-
 #[near_bindgen]
-impl News{
-    pub fn add(&mut self, hash_head: String, hash_body:String, uri: String){
-        
-        self.news.push(FN{
-            id: (self.news.len() as u32)+1, 
-            hash_head, 
-            hash_body, 
-            uri,
-            like: 0,
-            dislike: 0,
-        });
-    }
-
-    pub fn display_all(&self){
-        for i in self.news.iter(){
-            println!("{}. {}. Likes {}. Dislikes {}.", i.id, i.uri, i.like, i.dislike);
-        }
+impl NewsStorage
+{
+    pub fn add(&mut self, hash_head: String, hash_body:String, uri: String)
+    {
+        self.news.push(News
+            {
+                id: (self.news.len() as u32)+1, 
+                hash_head, 
+                hash_body, 
+                uri,
+                like: 0,
+                dislike: 0,
+            }
+        );
     }
 
-    pub fn display_by_index(&self, index: usize){
-        if (index) < self.news.len(){
-            println!("{}. {} . Likes {}. Dislikes {}.", 
-            self.news[index].id, 
-            self.news[index].uri,
-            self.news[index].like,
-            self.news[index].dislike);
-        }
+    pub fn get_all(&self)-> Vec<News>
+    {
+        self.news.clone()
     }
 
-    pub fn get_all(&self)-> Vec<FN>{
-        self.vec_clone()
-    }
-
-    pub fn get_by_index(&self, index: usize)-> FN{
+    pub fn get_by_index(&self, index: usize)-> News
+    {
         self.news[index].clone()
     }
 
-    pub fn vec_clone(&self) -> Vec<FN>{
-        let mut news_clone: Vec<FN>= Vec::new();
-        for i in self.news.iter(){
-            news_clone.push(i.clone());
-        }
-        news_clone
-    }
-
-    pub fn upvote(&mut self, index: usize) {
-        if index  < self.news.len(){
-        self.news[index ].like = self.news[index ].like.saturating_add(1);
-        }
-    }
-
-    pub fn downvote(&mut self, index: usize) {
-        if index < self.news.len(){
-        self.news[index ].dislike = self.news[index ].dislike.saturating_add(1);
-        }
-    }
-    pub fn get_like(&self, index: usize) -> u64 {
-        assert!(index < self.news.len());
-        self.news[index].like
-    }
-    pub fn get_dislike(&self, index: usize) -> u64 {
+    pub fn upvote(&mut self, index: usize) 
+    {
         assert!(index < self.news.len());
```

Comment:
please add meanigful assert desription like "index out of range"

https://doc.rust-lang.org/std/macro.assert.html#examples